storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -528,6 +529,19 @@ public Optional<Checkpoint> getJustifiedCheckpoint() {
     return dao.getJustifiedCheckpoint();
   }
 
+  @Override
+  public Optional<DepositTreeSnapshot> getFinalizedDepositSnapshot() {
+    return dao.getFinalizedDepositSnapshot();
+  }
+
+  @Override
+  public void setFinalizedDepositSnapshot(DepositTreeSnapshot finalizedDepositSnapshot) {
+    try (final HotUpdaterT updater = hotUpdater()) {
+      updater.setFinalizedDepositSnapshot(finalizedDepositSnapshot);
```

Comment:
We should remove deposits from before the snapshot from the database when it updates.

Reply:
But we always save it but load only when flag is toggled on (to enable one day, for example). Why we should force removal of deposits? We could remove it any day later